base-files: add reboot only button handler

alanswanson opened the pull request:
For devices such as BTHOMEHUBV5A with both reset and restart buttons, its easily accessible restart button has been assigned to KEY_POWER power script to poweroff preventing accidental (or malicious) factory resets by KEY_RESTART reset script. However an easily accessible button immediately powering off the device is also undesirable.

As KEY_RESTART is already used for reset script (and there's no KEY_REBOOT in Linux input events), use KEY_POWER2 for rebooting via new reboot script with 5 second seen delay.

Fixes FS#1965

ynezz commented on package/kernel/button-hotplug/src/button-hotplug.c:
```diff
@@ -85,6 +85,7 @@ static struct bh_map button_map[] = {
 	BH_MAP(BTN_9,		"BTN_9"),
 	BH_MAP(KEY_RESTART,	"reset"),
 	BH_MAP(KEY_POWER,	"power"),
+	BH_MAP(KEY_POWER2,	"reboot"),
```
This change doesn't correspond to the commit description (base-files package). It should go into separate commit.

ynezz commented on package/kernel/gpio-button-hotplug/src/gpio-button-hotplug.c:
```diff
@@ -100,6 +100,7 @@ static struct bh_map button_map[] = {
 	BH_MAP(KEY_LIGHTS_TOGGLE,	"lights_toggle"),
 	BH_MAP(KEY_PHONE,		"phone"),
 	BH_MAP(KEY_POWER,		"power"),
+	BH_MAP(KEY_POWER2,		"reboot"),
```
This change doesn't correspond to the commit description (base-files package). It should go into separate commit.

ynezz commented:
@alanswanson Ping.

alanswanson committed:
base-files: add reboot only button handler

For devices such as BTHOMEHUBV5A with both reset and restart buttons, its easily accessible
restart button has been assigned to KEY_POWER power script to poweroff preventing accidental
(or malicious) factory resets by KEY_RESTART reset script. However an easily accessible
button immediately powering off the device is also undesirable.

Fixes FS#1965
Signed-off-by: Alan Swanson <reiver@improbability.net>

alanswanson committed:
button-hotplug: add KEY_POWER2 handling

For devices such as BTHOMEHUBV5A with both reset and restart buttons, its easily accessible
restart button has been assigned to KEY_POWER power script to poweroff preventing accidental
(or malicious) factory resets by KEY_RESTART reset script. However an easily accessible
button immediately powering off the device is also undesirable.

As KEY_RESTART is already used for reset script (and there's no KEY_REBOOT in Linux input
events), use KEY_POWER2 for rebooting via new reboot script with 5 second seen delay.

Fixes FS#1965
Signed-off-by: Alan Swanson <reiver@improbability.net>

alanswanson committed:
gpio-button-hotplug: add KEY_POWER2 handling

For devices such as BTHOMEHUBV5A with both reset and restart buttons, its easily accessible
restart button has been assigned to KEY_POWER power script to poweroff preventing accidental
(or malicious) factory resets by KEY_RESTART reset script. However an easily accessible
button immediately powering off the device is also undesirable.

As KEY_RESTART is already used for reset script (and there's no KEY_REBOOT in Linux input
events), use KEY_POWER2 for rebooting via new reboot script with 5 second seen delay.

Fixes FS#1965
Signed-off-by: Alan Swanson <reiver@improbability.net>

alanswanson committed:
lantiq: BTHOMEHUBV5A: change restart button to use new KEY_POWER2 reboot button script

The restart button is currently assigned to KEY_POWER power script but an easily
accessible button immediately powering off the device is undesirable. Switch to using
new KEY_POWER2 reboot script with 5 second seen delay.

Fixes FS#1965
Signed-off-by: Alan Swanson <reiver@improbability.net>

alanswanson commented:
Sorry for the delay. Now split hotplug buttons changes into separate commits.

ynezz commented:
Thanks, I've just pushed it to master.
